api/rpc/server.go

```diff
@@ -0,0 +1,61 @@
+package rpc
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/filecoin-project/go-jsonrpc"
+)
+
+type Server struct {
+	http     *http.Server
+	rpc      *jsonrpc.RPCServer
+	listener net.Listener
+}
+
+func NewServer(address string, port string) *Server {
+	handler := jsonrpc.NewServer()
+	return &Server{
+		rpc: handler,
+		http: &http.Server{
+			Addr:    address + ":" + port,
+			Handler: handler,
+			// the amount of time allowed to read request headers. set to the default 2 seconds
+			ReadHeaderTimeout: 2 * time.Second,
+		},
+	}
+}
+
+// Start starts the RPC Server.
+func (s *Server) Start(context.Context) error {
+	listener, err := net.Listen("tcp", s.http.Addr)
+	if err != nil {
+		return err
+	}
+	s.listener = listener
+	log.Infow("RPC server started", "listening on", listener.Addr().String())
+	//nolint:errcheck
+	go s.http.Serve(listener)
```

Comment:
Looks like we are listening for just `tcp` here anyway. Maybe we can replace explicit listener here with  more convinient `ListenAndServe` and also don't store it in  the `Server` struct?